File: Problem2/gemini2.java
import java.util.Scanner;

public class gemini2 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        int n = input.nextInt();
        int[] nums = new int[n];
        
        // Use long instead of int to prevent integer overflow!
        long sum = 0; 
        
        for (int i = 0; i < n; i++) {
            nums[i] = input.nextInt();
            sum += nums[i];
        }

        if (sum % n == 0) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
        
        input.close();
    }
}

File: Problem2/gemini2.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;
    

    vector<int> nums(n);
    

    long long sum = 0;
    
    for (int i = 0; i < n; i++) {
        cin >> nums[i];
        sum += nums[i];
    }
    
    if (sum % n == 0) {
        cout << "YES" << endl;
    } else {
        cout << "NO" << endl;
    }
    
    return 0;
}